File: Array/Prefix Sum or Sum based approach/Minimum Number of Operations to Move All Balls to Each Box/Brute_Force_Approach.cpp
#include<bits/stdc++.h>
using namespace std;

//Time complexity :- O(N^2)
//Space complexity :- O(1)

//Approach :- 
// For every index traverse to the left and right of the array and find the distance of '1'  from that ind
// Dist is given by abs(j - i), where j is the index either on left or right of curr Index i

//Link :- https://leetcode.com/problems/minimum-number-of-operations-to-move-all-balls-to-each-box/description/

class Solution {
public:
    vector<int> minOperations(string boxes) {
        int n = boxes.size();
        vector<int> res(n, 0);

        for(int i=0; i<n; ++i){
            int lt = i-1, rt = i+1, cnt = 0;
            while(lt >= 0){
                if(boxes[lt] == '1')
                    cnt += (i - lt);
                
                lt--;
            }
            while(rt < n){
                if(boxes[rt] == '1')
                    cnt += (rt - i);
                
                rt++;
            }
            res[i] = cnt;
        }
        return res;
    }
};

File: Two Pointers/Fixed window/Largest 3-Same-Digit Number in String/brute_force_approach.ts

//Time complexity :- O(n*3)
//Space complexity :- O(n)

// Approach :-
// Try every subarray of 3 length if the are all equal then consider it as one of answer 


//link :- https://leetcode.com/problems/largest-3-same-digit-number-in-string/

function largestGoodInteger(num: string): string {
    let res = "";

    for(let i=0; i<num.length-2; ++i){
        let temp = num.substring(i, i+3);
        if(temp[0] === temp[1] && temp[1] === temp[2]){
            if(res === "" || (temp[0] > res[0]))
                res = temp;
        }
    }
    return res;
};